### PR TITLE
KOGITO-4588: DataType concept in the editor should cover both "custom" and "out-of-the-box" data types

### PR DESCRIPTION
**Pull request by vpellegrino**
Porting of https://github.com/kiegroup/kogito-editors-java/pull/196

### CODE REVIEW

**Comment by vpellegrino** [comment]
@ljmotta FYI, with https://github.com/kiegroup/kogito-tooling/pull/753/commits/3f978fbc3e508c2942809ee1f097e24399fabd4b, I created a fixed list of data types to be passed from the `DmnAutoTable` to its instance of the `BoxedExpressionProvider`.
With this PR, the `BoxedExpressionEditor` is always expecting, in input, a list of data types, that in the case of the DMN editor, is populated on the GWT side.
Please, let me know if this change sounds good to you or if you have a different solution in mind.

**Review by jomarko — APPROVED** [review]
Thank you for this backport.

**Comment by ljmotta** [comment]
@vpellegrino Thanks for the heads up. As far I saw, the dataTypes list is only used on the DataTypeSelector, which is not used on the DMN Runner Table. Taking this into account, it's not necessary to create a dataTypes array with the default values on the DmnAutoTable. You can pass an empty array to the dataTypes props of the BoxedExpressionProvider.

**Comment by vpellegrino** [comment]
> @vpellegrino Thanks for the heads up. As far I saw, the dataTypes list is only used on the DataTypeSelector, which is not used on the DMN Runner Table. Taking this into account, it's not necessary to create a dataTypes array with the default values on the DmnAutoTable. You can pass an empty array to the dataTypes props of the BoxedExpressionProvider.

Sure thing @ljmotta 
Just committed the required change

File: packages/boxed-expression-component/it-tests/integration/literal_expression_spec.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ describe("Literal Expression Tests", () => {
     cy.get(".literal-expression-header").click();
 
     cy.ouiaId("edit-expression-data-type").within(($container) => {
-      cy.get("input").click({ force: true });
+      cy.get("span.pf-c-select__toggle-text").click({ force: true });
     });
```
Comment (yesamer) [inline packages/boxed-expression-component/it-tests/integration/literal_expression_spec.ts:37]: @vpellegrino I found a mismatch with https://github.com/kiegroup/kogito-editors-java/pull/196/files#diff-6542fcaae434cd60b5a8922a54dc735d253f1d0fef1be2ece2f320015f90a431L24, is this intended?

File: packages/boxed-expression-component/tests/components/Table/Table.test.tsx [hunk 2]
```diff
@@ -230,7 +230,9 @@ describe("Table tests", () => {
       expect(baseElement.querySelector(EXPRESSION_POPOVER_MENU)).toBeTruthy();
       expect(baseElement.querySelector(EXPRESSION_POPOVER_MENU_TITLE)?.innerHTML).toBe(editRelationLabel);
       expect((baseElement.querySelector(EDIT_EXPRESSION_NAME)! as HTMLInputElement).value).toBe(columnName);
-      expect((baseElement.querySelector(EDIT_EXPRESSION_DATA_TYPE)! as HTMLInputElement).value).toBe(DataType.Boolean);
+      expect((baseElement.querySelector(EDIT_EXPRESSION_DATA_TYPE)! as HTMLSpanElement).textContent).toBe(
+        DataType.Boolean
+      );
     });
```
Comment (yesamer) [inline packages/boxed-expression-component/tests/components/Table/Table.test.tsx:237]: @vpellegrino Another mismatch here https://github.com/kiegroup/kogito-editors-java/pull/196/files#diff-417808584575362ce894ea2e037b441a08f5ccd7bcd9a5d7fe1d79c5359c1233L21, is `Column` used here?

**Review by yesamer — APPROVED** [review]
Thank you for porting this @vpellegrino. I just checked the difference with the original PR, it looks correct with a couple of minor differences which are probably intended.

File: packages/boxed-expression-component/it-tests/integration/literal_expression_spec.ts (hunk 1 above)
Comment (vpellegrino) [reply packages/boxed-expression-component/it-tests/integration/literal_expression_spec.ts:37]: yes, test was failing

File: packages/boxed-expression-component/tests/components/Table/Table.test.tsx (hunk 2 above)
Comment (vpellegrino) [reply packages/boxed-expression-component/tests/components/Table/Table.test.tsx:237]: image
we have the same change also in the ported PR

**Comment by vpellegrino** [comment]
> Thank you for porting this @vpellegrino. I just checked the difference with the original PR, it looks correct with a couple of minor differences which are probably intended.

There are several differences with the original ported PR, because of the different folder structure.
AFAIK, the changes are all for the testing code (not the production one).

File: packages/boxed-expression-component/tests/components/Table/Table.test.tsx (hunk 2 above)
Comment (yesamer) [reply packages/boxed-expression-component/tests/components/Table/Table.test.tsx:237]: I meant the import of `Column` at line 21  was removed in the original PR and not in the current one.

Comment (vpellegrino) [reply packages/boxed-expression-component/tests/components/Table/Table.test.tsx:237]: image
It seems there was (is) no `Column` in the imports.

**Comment by yesamer** [comment]
@vpellegrino Thank you!

@ljmotta @tiagobento Can we proceed to merge this?

**Review by ljmotta — APPROVED** [review]
@vpellegrino @yesamer It looks good to me!

**Pull request merged**